### ISSUE TITLE
Support Struct datatype in assert DataFrame/Series equal

### ISSUE THREAD

**Issue by physinet** [pola-rs/polars#3186]
The methods `pl.testing.assert_frame_equal` and `pl.testing.assert_series_equal` do not appear to support `DataFrame` or `Series` objects with struct types:
```
>>> import polars as pl
>>> s = pl.DataFrame([{"b": 1}]).to_struct("a")
>>> pl.testing.assert_series_equal(s, s)
thread '<unnamed>' panicked at 'not implemented', /github/workspace/polars/polars-core/src/series/comparison.rs:183:17
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/vadmin/.venv-dev/lib/python3.10/site-packages/polars/testing.py", line 99, in assert_frame_equal
    _assert_series_inner(
  File "/home/vadmin/.venv-dev/lib/python3.10/site-packages/polars/testing.py", line 182, in _assert_series_inner
    if (left != right).sum() != 0:
  File "/home/vadmin/.venv-dev/lib/python3.10/site-packages/polars/internals/series.py", line 332, in __ne__
    return self._comp(other, "neq")
  File "/home/vadmin/.venv-dev/lib/python3.10/site-packages/polars/internals/series.py", line 321, in _comp
    return wrap_s(getattr(self._s, op)(other._s))
pyo3_runtime.PanicException: not implemented
```
The same exception is raised for `DataFrames`:
```
>>> df = s.to_frame()
>>> pl.testing.assert_frame_equal(df, df)
```
This capability is essential for unit testing with `DataFrame`s containing structs. Although unnesting the struct fields before applying the assert functions could be a workaround, I think users could benefit from automatic comparison of struct fields.

**Comment by physinet** [comment]
@cjermain has made progress on this in #3180

**Comment by ritchie46** [comment]
Fixed by @cjermain in #3180
